symengine/tests/cwrapper/test_cwrapper.c

```diff
@@ -458,6 +459,213 @@ void test_ascii_art() {
     basic_str_free(s);
 }
 
+void test_functions() {
+    basic pi, e;
+    basic minus_one, minus_half, zero, one, two, four;
+    basic pi_div_two,pi_div_four;
+    basic e_minus_one;
+    basic ans;
+
+    basic_new_stack(pi);
+    basic_new_stack(e);
+    basic_new_stack(ans);
+    basic_new_stack(two);
+    basic_new_stack(pi_div_two);
+    basic_new_stack(four);
+    basic_new_stack(pi_div_four);
+    basic_new_stack(one);
+    basic_new_stack(minus_one);
+    basic_new_stack(zero);
+    basic_new_stack(e_minus_one);
+    basic_new_stack(minus_half);
+
+    basic_const_pi(pi);
+    basic_const_E(e);
+    integer_set_si(two, 2);
+    integer_set_si(four, 4);
+    integer_set_si(one, 1);
+    integer_set_si(minus_one, -1);
+    integer_set_si(zero, 0);
+    
+    basic_div(pi_div_two, pi, two);
+    basic_div(pi_div_four, pi, four);
+    basic_pow(e_minus_one, e, minus_one);
+    basic_mul(e_minus_one, e_minus_one, minus_one);
+    basic_div(minus_half, minus_one, two);
+
+    char* s;
+
+    basic_sin(ans, pi);
+    s = basic_str(ans);
+    SYMENGINE_C_ASSERT(strcmp(s, "0") == 0);
```

Comment:
You won't need these `strcmp`s at all. You are already comparing them with `basic_eq`.